packages/dx-scheduler-core/src/plugins/drag-drop-provider/helpers.ts

```diff
@@ -63,46 +65,84 @@ export const autoScroll: PureComputed<
   }
 };
 
-export const calculateAppointmentTimeBoundaries: CalculateAppointmentTimeBoundaries = (
+export const timeBoundariesByResize: TimeBoundariesByResize = (
+  payload, targetData, targetType, cellDurationMinutes, insidePart,
+) => {
+  if (targetType !== payload.appointmentType) {
+    return { appointmentStartTime: undefined, appointmentEndTime: undefined };
+  }
+
+  let appointmentStartTime;
+  let appointmentEndTime;
+  const sourceType = payload.type;
+
+  if (sourceType === RESIZE_TOP) {
+    const insideTopOffset = targetType === VERTICAL_TYPE
+      ? insidePart * cellDurationMinutes * 60 / 2 : 0;
+    appointmentStartTime = moment(targetData.startDate as Date)
+      .add(insideTopOffset, SECONDS).toDate();
+    appointmentEndTime = moment(payload.endDate as Date).toDate();
+  }
+  if (sourceType === RESIZE_BOTTOM) {
+    const insideBottomOffset = insidePart === 0 && targetType === VERTICAL_TYPE
+      ? cellDurationMinutes * 60 / 2 : 0;
+    appointmentEndTime = moment(targetData.endDate as Date)
+      .add(-insideBottomOffset, SECONDS).toDate();
+    appointmentStartTime = moment(payload.startDate as Date).toDate();
+  }
+  // keep origin appointment duration if coordinates are wrong
+  if (moment(appointmentEndTime).diff(appointmentStartTime, MINUTES) < 1) {
+    appointmentStartTime = moment(payload.startDate as Date).toDate();
```

Comment:
Does not `moment(date).toDate()` equal `date`?

Reply:
Yes, it does.